har/entrylist.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2020 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package har
+
+import (
+	"container/list"
+	"sync"
+)
+
+type EntryList struct {
+	lock  sync.Mutex
+	Items *list.List
+}
+
+func NewEntryList() *EntryList {
+	return &EntryList{
+		Items: list.New(),
+	}
+}
+
+func (el *EntryList) Lock() {
+	el.lock.Lock()
+}
+
+func (el *EntryList) Unlock() {
+	el.lock.Unlock()
+}
```

Comment:
Why do we need to expose the lock/unlock out of the EntryList struct? The functions on EntryList can work with el.lock directly

Reply:
So the thought was that if someone wanted to create a particular variant of a har file - for example, someone wanted all hars that had a URL of "*.google.com" - they would need to be able to iterate the list. Without the lock being exposed that would end up non goroutine safe.

However, my problem may very well be that my brain is still stuck at times in "ruby mode" where one can sort of stick their toe in and out of an object/struct as they please. I'm thinking the better option to handle my thought process would be for a container interface that one could implement in any way they wished outside of martian. That would allow for the locks to be unexported because the internal implementation would be solely for the purposes of martian itself and if one wanted the "*.google.com" concept they would implement a new container interface and set the logger to use that. 

In fact nothing would need to be exported from martian but rather if martian allowed the container interface implementation to be set then the caller would have complete control of the container outside of martian.

Would your preference be to close this pull or to rework within this request?

Thanks for your patience on this......

Reply:
I decided to just add another commit to this pull request because the difference was not that much upon adding an interface.